docs/proposals/hosted_components_templates.md

```diff
@@ -0,0 +1,559 @@
+# Use [OCP templates and] APBs to install hosted components
+
+## Description
+Currently most of our hosted components have their own repositories where they
+maintain changes and then need to synchronize those changes within their
+respective roles. This can cause delays and times where features don't
+match up with what is installed or can be configured.
+
+There is also a concern that copying files from one location to another can
+cause cases where files are out of sync or versions don't align with what is
+installed and what is configured (is what we configured correct for the image
+version we installed).
+
+## Rationale
+As was piloted with the Template Service Broker ("TSB" for short), a role was
+created to process and configure OCP templates that were synchronized (currently
+still a manual process) from the origin repo where the component resides.
+
+This allowed any changes for the installation to be maintained by the developers
+and keep them from needing to be fluent with Ansible. It also afforded a level
```

Comment:
I'm a little unclear if this proposal is focused on something similar to TSB which would isolate devs from Ansible largely focusing just on templates and params, or an APB which would contain playbooks and roles maintained by the developers. (or is it both, as necessary?)

It seemed like our hosted applications were probably more complex than just a template + params, but a good fit for full APBs, though devs would still need to deal with ansible.

Reply:
I need to clarify that because based on Clayton's comments before that seems unclear since I talk about both. The intent is that the deliverable would be an APB, however within those roles/playbooks devs could use the model the TSB takes to isolate them from Ansible.

I agree, there will need to be work with Ansible for some components as they don't lend well to being contained within a template object, however if there are template objects already being provided by the team, then it would make sense that they leverage and reuse those rather than duplicating their logic into a module while also maintaining a template file.